sdk/metric/manual_reader.go

```diff
@@ -43,6 +44,11 @@ func NewManualReader() Reader {
 func (mr *manualReader) register(p producer) {
 	mr.lock.Lock()
 	defer mr.lock.Unlock()
+	if mr.producer != nil {
```

Comment:
Since register() is a private interface, it seems the only reason this could fail is due to an internal bug. The way this test is written, it looks more like we're checking for user-level misbehavior. Another way of stating this is: what could the user possibly do if they see this error?

Reply:
This can fail if a user registers a reader twice. E.g.

```
r := NewManualReader()
_ := NewMeterProvider(WithReader(r))
_ := NewMeterProvider(WithReader(r))
```

The second call will try to register reader r after it was registered in the first call. This second registration cannot be done according to the specification. Dropping the registration attempt is in line with the specification, but would not match user expectation in the situation. Logging an error describing the failure should help the user debug here.